### PR TITLE
support theme param in links

### PR DESCRIPTION
**Pull request by OriHoch**

### CODE REVIEW

**Comment by OriHoch** [comment]
test links - 
http://localhost:8000/budget/0033270712/2006?li=0
http://localhost:8000/org/company/520040205?li=0
http://localhost:8000/org/association/580002483?li=2
http://localhost:8000/national-budget-changes/2006/33-019?li=0
http://localhost:8000/supports/0036100202/2011/%D7%94%D7%9E%D7%9B%D7%9C%D7%9C%D7%94%20%D7%94%D7%98%D7%9B%D7%A0%D7%95%D7%9C%D7%95%D7%92%D7%99%D7%AA%20%D7%91%D7%90%D7%A8%20%D7%A9%D7%91%D7%A2%20-%20%D7%A7%D7%93%D7%99%D7%9E%D7%94%20%D7%9E/%D7%AA%D7%9E%D7%99%D7%9B%D7%94%20%20%D7%93%D7%A8%D7%9A%20%D7%9E%D7%95%D7%A1%D7%93?li=0
http://localhost:8000/tenders/exemptions/536224/none?li=0
http://localhost:8000/contract-spending/4501519850/0017310901?li=3

File: app/pipes/render-template.ts [hunk 1]
```diff
@@ -27,11 +27,18 @@ env.addFilter('split', function(x: string) {
   }
 });
 
+env.addFilter('search_link', function(urlEncodedParams: string, theme_id: string) {
+  return '//next.obudget.org/s/?' + urlEncodedParams + (theme_id ? '&theme=' + theme_id : '');
+});
 
+env.addFilter('item_link', function(doc_type, entity_kind, entity_id, theme_id: string) {
```
Comment (akariv) [inline app/pipes/render-template.ts]: It's better to use the document id here and not take the entity_kind, entity_id parameters which are only relevant for entities.

File: app/pipes/render-template.ts [hunk 2]
```diff
@@ -27,11 +27,18 @@ env.addFilter('split', function(x: string) {
   }
 });
 
+env.addFilter('search_link', function(urlEncodedParams: string, theme_id: string) {
```
Comment (akariv) [inline app/pipes/render-template.ts]: it's better here to get the term as parameter and not encoded query params

**Comment by OriHoch** [comment]
@akariv ready for review

**Comment by akariv** [comment]
@OriHoch ^

**Comment by OriHoch** [comment]
@akariv thanks, fixed

File: app/descriptors/budget/budget2dig/subtitle-template.html [hunk 3]
```diff
@@ -1,5 +1,5 @@
 סעיף תקציבי בתקציב המדינה - תחת תת-הנושא
-<a href="/i/budget/C{{(func_cls_json[0] | load_json)[0]}}{{(func_cls_json[0] | load_json)[2]}}/{{year}}">
+<a href="{{ 'budget' | item_link('C'+(func_cls_json[0] | load_json)[0]+(func_cls_json[0] | load_json)[2] + '/' + year, theme_id) }}">
```
Comment (akariv) [inline app/descriptors/budget/budget2dig/subtitle-template.html]: do we need to add theme_id every time? 😢

Comment (OriHoch) [reply app/descriptors/budget/budget2dig/subtitle-template.html]: I had some problems due to the way these templates are rendered, I could try again

Comment (akariv) [reply app/descriptors/budget/budget2dig/subtitle-template.html]: since it's a global value in the app, it should be possible to avoid it I think

**Comment by OriHoch** [comment]
@akariv fixed
also, fixed to support theme param in redirect

**Comment by OriHoch** [comment]
@akariv regarding theme param in redirect - I assume it will always be an item page URL without theme param..

**Pull request merged**